src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,206 @@
+        if (selection == 0) {
+            FileUtils.clearDirectory(BACKPAGE_PATH, EXECUTABLE_NAME);
+            FileUtils.createDirectories(BACKPAGE_PATH);
+
+            downloadBackpage(progressBar, asset);
+            Files.writeString(VERSION_PATH, remote.toString());
+            return true;
+        }
+
+        return current != null;
+    }
+
+    private void downloadBackpage(JProgressBar progressBar, ReleaseInfo.Asset asset) throws IOException {
+        SwingUtilities.invokeLater(() -> progressBar.setIndeterminate(false));
+
+        try (ZipInputStream zis = new ZipInputStream(new BufferedInputStream(asset.openStream()))) {
+            int downloadedBytes = 0, progress = 0, partSize = asset.size / 50;
+
+            for (ZipEntry entry = zis.getNextEntry(); entry != null; entry = zis.getNextEntry()) {
+                Path to = BACKPAGE_PATH.resolve(entry.getName());
+
+                if (entry.isDirectory()) {
+                    Files.createDirectories(to);
+                    continue;
+                }
+
+                File file = to.toFile();
+                try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(file))) {
+                    double compressionRatio = (double) entry.getCompressedSize() / entry.getSize();
+
+                    int read;
+                    byte[] buffer = new byte[8192];
+                    while ((read = zis.read(buffer)) != -1) {
+                        bos.write(buffer, 0, read);
+
+                        downloadedBytes += (read * compressionRatio);
+                        int currentProgress = downloadedBytes / partSize;
+                        if (progress != currentProgress)
+                            setProgress(progressBar, progress = currentProgress);
+                    }
+
+                    if (OSUtil.isLinux() && file.getName().startsWith(EXECUTABLE_NAME))
+                        file.setExecutable(true);
+                }
+            }
+        }
+    }
+
+    private void setProgress(JProgressBar progressBar, int progress) {
+        SwingUtilities.invokeLater(() -> progressBar.setValue(progress));
+    }
+
+    private @Nullable Version readVersionFile() throws IOException {
+        if (Files.notExists(VERSION_PATH)) return null;
+        return new Version(Files.readString(VERSION_PATH));
+    }
+
+    public static class ReleaseInfo {
+        public String name;
+        public List<Asset> assets;
+
+        public static ReleaseInfo get() throws IOException, JsonParseException {
+            return new Gson().fromJson(Http.create(RELEASE_URL).getContent(), ReleaseInfo.class);
+        }
+
+        public Version getVersion() {
+            return new Version(name);
+        }
+
+        public Asset getValidAsset() {
+            return assets.stream()
+                    .filter(ReleaseInfo.Asset::isSupported)
+                    .findFirst().orElse(null);
+        }
+
+        public static class Asset {
```

Comment:
i'd advise against nesting this inside release info, BackpageTask.ReleaseInfo.Asset is just... not ideal, it can be just BackpageTask.Asset